omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/EncounterResource1_9.java

```diff
@@ -36,25 +39,47 @@ public DelegatingResourceDescription getRepresentationDescription(Representation
 		if (rep instanceof DefaultRepresentation) {
 			DelegatingResourceDescription description = super.getRepresentationDescription(rep);
 			description.addProperty("visit", Representation.REF);
+            description.removeProperty("provider");
+            description.addProperty("encounterProviders", Representation.DEFAULT);
 			return description;
 		} else if (rep instanceof FullRepresentation) {
 			DelegatingResourceDescription description = super.getRepresentationDescription(rep);
 			description.addProperty("visit", Representation.DEFAULT);
+            description.removeProperty("provider");
+            description.addProperty("encounterProviders",Representation.DEFAULT);
 			return description;
 		}
 		return null;
 	}
-	
-	/**
+
+    @Override
+    public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+        DelegatingResourceDescription description = super.getUpdatableProperties();
+        description.addProperty("encounterProviders");
+        return description;
+    }
+
+    /**
 	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
 	 */
 	@Override
 	public DelegatingResourceDescription getCreatableProperties() {
 		DelegatingResourceDescription description = super.getCreatableProperties();
 		description.addProperty("visit");
+        description.addProperty("encounterProviders");
 		return description;
 	}
-	
+
+    @Override
+    public Encounter save(Encounter delegate) {
+        //This is a hack to save encounterProviders correctly. Without this they are created without encounter_id in
+        //the database.
+        for(EncounterProvider ep:delegate.getEncounterProviders()){

```

Comment:
btw, please format this nicely in the OpenMRS style, e.g. spaces like

    for (EncounterProvider ep : delegate.getEncounterProviders()) {